components/wealthbox/sources/common/common.mjs

```diff
@@ -0,0 +1,82 @@
+import wealthbox from "../../wealthbox.app.mjs";
+import { DEFAULT_POLLING_SOURCE_TIMER_INTERVAL } from "@pipedream/platform";
+
+export default {
+  props: {
+    wealthbox,
+    db: "$.service.db",
+    timer: {
+      type: "$.interface.timer",
+      default: {
+        intervalSeconds: DEFAULT_POLLING_SOURCE_TIMER_INTERVAL,
+      },
+    },
+  },
+  hooks: {
+    async deploy() {
+      const historicalEvents = await this.getEvents({
+        params: {
+          per_page: 25,
+        },
+      });
+      if (!(historicalEvents?.length > 0)) {
+        return;
+      }
+      this._setLastCreated(this.getCreatedAtTs(historicalEvents[0]));
+      historicalEvents.forEach((event) => this.emitEvent(event));
+    },
+  },
+  methods: {
+    _getLastCreated() {
+      return this.db.get("lastCreated");
+    },
+    _setLastCreated(lastCreated) {
+      this.db.set("lastCreated", lastCreated);
+    },
+    getCreatedAtTs(event) {
+      return (Date.parse(event.created_at)) / 1000;
+    },
+    emitEvent(event) {
+      const meta = this.generateMeta(event);
+      this.$emit(event, meta);
+    },
+    getEvents() {
+      throw new Error("getEvents is not implemented");
+    },
+    generateMeta() {
+      throw new Error("generateMeta is not implemented");
+    },
+  },
+  async run() {
+    const lastCreated = this._getLastCreated() || 0;
+    let maxLastCreated = lastCreated;
+    let total;
+    const params = {
+      per_page: 25,
+      page: 1,
```

Comment:
you aren't incrementing the page, is that right?

Reply:
Oops, thanks. :)